swarm/CHANGELOG.md

```diff
@@ -1,3 +1,15 @@
+# 0.40.1 - [unreleased]
+
+- Add new `on_swarm_event` method to `NetworkBehaviour` that accepts a `FromSwarm` enum and update
+  `inject_*` methods to call `on_swarm_event` with the respective `FromSwarm` variant and deprecate
+  them. See [PR 3011].
```

Comment:
Can you add an additional paragraph telling users what they need to do, e.g. something along the lines of:

> To migrate, replace your `NetworkBehaviour::inject_*` calls with a single implementation of `NetworkBehaviour::on_swarm_event`.

(My wording is likely not ideal.)